libs/features/webapp-prompt/webapp-prompt.js

```diff
@@ -35,6 +35,16 @@ const getIcon = (content) => {
   return icons.company;
 };
 
+const modalsActive = () => !!document.querySelector('.dialog-modal');
+
+const waitForClosedModalsThen = (loadPEP) => {
+  if (modalsActive()) {
+    setTimeout(() => waitForClosedModalsThen(loadPEP), 200);
```

Comment:
Could we not use a `setInterval` to avoid the recursive calls?

Reply:
I'd prefer to not use setInterval since we'd have to keep track of the `setInterval` ID and clean it up after with `clearInterval`. And `waitForClosedModalsThen` isn't recursive in the sense that it's growing the callstack indefinitely, it, like `setInterval`, adds a function to the task queue, and in that sense is (to us, at least) indistinguishable in behavior from `setInterval`. The question is of whether we prefer the extra code to clean up the interval or the extra code for adding to the event queue.